src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -1982,7 +1994,7 @@ class C : B
         End Function
 
         <WorkItem(545967, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545967")>
-        <MemberData(NameOf(AllCompletionImplementations))>
+        <InlineData(CompletionImplementation.Legacy)> ' Up and Down keys are supported in VSSDK tests
```

Comment:
why is this not tested against modern?

Reply:
The Editor does not provide us capabilities to test Up and Down key presses in the completion popup. We opened an issue on them to implement this and similar tests on their side.

---
In reply to: [222523062](https://github.com/dotnet/roslyn/pull/29016#discussion_r222523062) [](ancestors = 222523062)

Reply:
The up\down interface in Roslyn tests
1) Calls Roslyn's up\down handler
2) Invokes editor operations for moving the caret up\down

It does not exercise the _modern commanding_ codepath in the editor, which
1) Calls modern command handlers, including the one of async completion
2) Invokes editor operations for moving the caret up\down

Reply:
SendUp and SendDown are not available calling in Roslyn tests.

Those tests should be implemented on the editor side: [https://github.com/dotnet/roslyn/issues/30122](https://github.com/dotnet/roslyn/issues/30122)

---
In reply to: [223930069](https://github.com/dotnet/roslyn/pull/29016#discussion_r223930069) [](ancestors = 223930069)

Reply:
Added tests for up and down. I hope we have now a test parity except cases covered by GitHub issues.

---
In reply to: [229259166](https://github.com/dotnet/roslyn/pull/29016#discussion_r229259166) [](ancestors = 229259166)